[WNMGDS-2020] updates for hint text from DSM (#2249)
CMSgov/design-system: packages/docs/content/components/form-label.mdx

## Guidance

### Form legend

- Use a single legend for fieldset (this is required). One example of a common use of fieldset and legend is a question with radio button options for answers. The question text and radio buttons are wrapped in a fieldset, with the question itself being inside the legend tag.

> - Avoid links in legends when possible. Links inside legends are not read by some screen readers, most notably Internet Explorer 11, paired with JAWS. If you need a pattern like this, put the link outside of the legend element, but keep all other useful hint text in the legend.

Reviewer: Is this valid now that we no longer support IE 11? @SparrowSpit, do you know of any other software combos that would make this necessary? Or is it possibly a bad pattern in general?

Reviewer 2: [It looks like this is a known issue](https://github.com/FreedomScientific/VFO-standards-support/issues/402) and I also tested and experienced the issue described in that convo. 

> You tab into a button within the element. The legend button is read.
You tab into a button underneath the contained within the fieldset. The fieldset button is read.
You shift-tab back to the button within the element.
The previous button is read back to you.

Reviewer: Okay, so it's also a problem for Chrome, Firefox, and Edge according to that ticket. Thanks, @SparrowSpit! Maybe we should update it to say 
```suggestion
- Avoid links in legends when possible. Links inside legends are not read by some screen readers, namely JAWS. If you need a pattern like this, put the link outside of the legend element, but keep all other useful hint text in the legend.
```

### Form label

- Each form field should have a `<label>`. Never use a field's placeholder attribute as the primary way to label the field.
- Each form field should have a clear, brief, and descriptive label.
- Labels should have a `for` attribute, referencing the corresponding input's unique `id` attribute. Only one label can be associated to each unique form element.
- Labels should be placed above their fields.
- Label text should be short and in sentence case.
- Avoid colons at the end of labels.
- Don't use a field's placeholder attribute as the primary way to label the field.

#### Content

- Describe what the user needs to do when the field is selected
- Use a phrase or complete sentence
- Start with an action verb and pair it with a noun to help it be clearer. Example: Enter your age.

> - Use the least amount of words possible (i.e. don't use "please")

Reviewer: ```suggestion
- Use the fewest words possible (i.e. don't use "please")
```

Reviewer 3: Does this bullet point contradict the one 2 points before (use a phrase or complete sentence)?

Do we care if folks to use numerically fewer words, or are we asking them to use direct/active language when writing labels?

Also, the "i.e." here should probably be "e.g." as "please" is being used as an example.

- Use sentence case
- Be consistent with punctuation at the end of a phrase or sentence
- Be consistent with phrase and sentence structure

### Hint text

- Place hint text within the field's `<label>` element.
- Place hint text where appropriate depending on what content it's helping to explain
- Use hint text for supporting contextual help, which will always be shown.

> - Hint text should sit between a from label and the input.

Reviewer: ```suggestion
- Hint text should sit between a form label and the input.
```

- Use lower visual style (i.e., lighter gray than the form label or the input option) so it won't compete for users' attention with the field label. 
- Use hint text to indicate which inputs are optional when there are more required fields than optional.
- Hint text can also appear after an input box when the text is being used to show the value someone should be entering in the field (e.g., income amount input box displays text after the input box to highlight the amount should be "monthly income".)

#### Content

- Use when appropriate to support a form label (Goes below the form label in a lighter weight font)
- Describe how to best answer the question/form label

> - Use the least amount of words possible (i.e. don't use "please")

Reviewer: ```suggestion
- Use the fewest words possible (i.e. don't use "please")
```
